### PR TITLE
ecshreve: OpenAPI documentation

### PR DESCRIPTION
**Pull request by ecshreve**
## What does this do?
\<It's not clear if I don't update this text with relevant info\>
This PR does a few things
- Creates an OpenAPI 3.0 definition for the API
- Add `npm` tasks to validate and bundle the definition
- Update `npm build` to also handle bundling the definition and copying it to the `/dist ` directory
- Add page to host the new documentation rendered with rapidoc
- Add link to new docs on the existing docs page

one note: the new docs don't include query parameters yet, I'm working on them this evening

Much more detail and context can be found in the `swagger/README.md` file.

## How can I try this out?
Pull this branch, run `npm start`, navigate to `http://localhost:3000/openapidocs`

demo:
oapi3

## How was it tested?
localhost / manual validation

## Is there a Github issue this is resolving?
no

## Here's a fun image for your troubles
random photo - update me

### CODE REVIEW

**Comment by ecshreve** [comment]
whew! alrighty at this point the changes in this PR should cover everything documented in the existing docs, with a whole lotta other functionality on top of it. let me know if ya have any questions!

File: package.json [hunk 1]
```diff
@@ -68,4 +71,4 @@
     "typedoc": "^0.22.12",
     "typescript": "^4.5.5"
   }
-}
+}
```
Comment (bagelbits) [inline package.json:74]:
```suggestion
}

```

File: src/swagger/parameters/path/languages.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+name: index
+in: path
+required: true
+description: |
+  The `index` of the language to get.
+schema:
+  type: string
+  enum:
```
Comment (bagelbits) [inline src/swagger/parameters/path/languages.yml:8]: I like how  you enum the shorter lists.

**Review by bagelbits — DISMISSED** [review]
Looks  great!

File: src/swagger/parameters/path/languages.yml (hunk 2 above)
Comment (ecshreve) [reply src/swagger/parameters/path/languages.yml:8]: thanks! i figured anything with less than 30 options might as well put it in an enum

**Comment by ecshreve** [comment]
@bagelbits did you mean to approve this? looks like maybe you dismissed it earlier after resolving that comment on the package.json file?

**Comment by bagelbits** [comment]
Huh. How did that happen.

**Review by bagelbits — APPROVED** [review]
LGTM

**Pull request merged**